### ISSUE TITLE
NumberField: userEvent.click on increment/decrement in storybook does not work anymore

### ISSUE THREAD

**Issue by vincent-capicotto-payfit** [adobe/react-spectrum#9406]
### Provide a general summary of the issue here

Since react-aria-components 1.14.0, my play functions in storybook does not work anymore.
If I click manually on increment/decrement buttons it works, but not with the userEvent.click function, the button are found but the value does not change

### 🤔 Expected Behavior?

As before, be able to test programmatically the number field increment and decrement using userEvent from storybook/test

### 😯 Current Behavior

The click on increment and decrement does nothing when userEvent.click is triggered

### 🖥️ Steps to Reproduce

N/A

### Version

1.14.0

### What browsers are you seeing the problem on?

Chrome

### What operating system are you using?

IOS / Ubuntu

**Comment by gylxan** [comment]
We have the same error with the `userEvent` of `@testing-library/user-event` in our test environment with `vitest`.

**Comment by snowystinger** [comment]
@chirokas your PR says Android TalkBack, but this issue seems would seem to indicate this is only for test environments. Can anyone provide reproductions details for the Android TalkBack? I'll try to look soon.

If it's just test environments, then they (the tests) should be updated to use the correct sequence of events, we shouldn't be adding code specific to testing if possible.

**Comment by chirokas** [comment]
> Can anyone provide reproductions details for the Android TalkBack?

@snowystinger

Go to S2 NumberField in the docs and press the increment/decrement buttons

https://github.com/user-attachments/assets/b3b3b78d-42e6-405e-9541-e8ac438841d9

**Comment by snowystinger** [comment]
Thanks, I was able to verify the bug and the fix.

Has anyone verified that this addresses the the issues with the tests? @gylxan @vincent-capicotto-payfit  ?

**Comment by gylxan** [comment]
Looks good from my point of view! Thanks for your quick fix!

**Comment by vincent-capicotto-payfit** [comment]
I didn't test it, I will when it will be released, but it seems good
